diff -ruN a/packages/kokkos/core/src/impl/Kokkos_MemoryPool.cpp b/packages/kokkos/core/src/impl/Kokkos_MemoryPool.cpp
--- a/packages/kokkos/core/src/impl/Kokkos_MemoryPool.cpp	2024-02-27 14:49:58.707067651 +0100
+++ b/packages/kokkos/core/src/impl/Kokkos_MemoryPool.cpp	2024-02-27 14:52:38.362774953 +0100
@@ -46,6 +46,7 @@
 
 #include <ostream>
 #include <sstream>
+#include <cstdint>
 
 //----------------------------------------------------------------------------
 //----------------------------------------------------------------------------
